### PR TITLE
Remove charUnHex

### PR DESCRIPTION
**Pull request by a-noni-mousse**
### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

File: src/ripple/basics/StringUtilities.h [hunk 1]
```diff
@@ -48,6 +48,24 @@ template <class Iterator>
 std::optional<Blob>
 strUnHex(std::size_t strSize, Iterator begin, Iterator end)
 {
+    static constexpr std::array<int, 256> const unxtab = []() {
```
Comment (scottschurr) [inline src/ripple/basics/StringUtilities.h:51]: This file now needs a `#include <array>`.  Please add that near the top of the file like this...
```
...
#include <boost/utility/string_view.hpp>
#include <array>
#include <optional>
...
```

File: src/ripple/basics/StringUtilities.h [hunk 2]
```diff
@@ -56,25 +74,23 @@ strUnHex(std::size_t strSize, Iterator begin, Iterator end)
...
 
         out.push_back(c);
-        ++iter;
     }
 
     while (iter != end)
     {
-        int cHigh = charUnHex(*iter);
+        int cHigh = unxtab[*iter++];
         ++iter;
```
Comment (scottschurr) [inline src/ripple/basics/StringUtilities.h:88]: By accident you left in a double increment.  Fortunately it causes massive unit test failures.  Notes:
1. You'll need to remove the `++iter;` from this line.
2. Consider running the unit tests before submitting pull requests in the future.  It's easy.  Build rippled, then invoke...
```
$ <path to rippled> --unittest
```
Then wait for the results.  It takes a while, but if it says `0 failures` at the end then you're less likely to have introduced a bug.

**Review by scottschurr — CHANGES_REQUESTED** [review]
Thanks for the submission.  This pull request looks pretty close to right, but needs a couple of fixes.  Once those fixes are in place I think this will be good to go.

**Comment by cryptobrad** [comment]
Thanks for looking closely
I fixed the bug you found

**Review by scottschurr — APPROVED** [review]
:+1: Latest version looks good to me!  Thanks.

**Comment by nbougalis** [comment]
Merged as 83ac141f656b1a95b5661853951ebd95b3ffba99

**Pull request closed without merging**